\documentclass[10pt]{book}
\bibliographystyle{apalike}

\input{defs.tex}

\begin{document}
\title{BioE241 labs}
\author{Ian Holmes \\ 374C Stanley Hall \\ UC Berkeley}
\date{}

\maketitle

\titlecite

\tableofcontents

\chapter{Introduction}

This handout describes a series of labs for BioE241,
accompanying the theoretical lectures in that class,
which describe various statistical models for biological data.

The class has a (non-exclusive) emphasis on models that describe the evolution of
DNA, RNA and amino acid sequences on phylogenetic trees.

Other software tools are also used.
The labs also ask you to develop some pseudocode and actual implementations
(generally in a programming language of your choice)
and to do a small amount of elementary math.

\section{Useful URLs}

\begin{tabular}{ll}
BioE241 class homepage
 & \url{http://biowiki.org/BioE241} \\
\\
DART software homepage
 & \url{http://biowiki.org/DART} \\
 (used by most of the labs) & \\
\\
Class materials repository
 & \url{https://github.com/ihh/bioe241} \\
 (lecture notes, these lab handouts, data files) & \\
\\
DART source code repository
 & \url{https://github.com/ihh/dart} \\
\end{tabular}

% Include individual labs
\input{simulation/lab.tex}
\input{search/lab.tex}
\input{phylogeny/lab.tex}
\input{alignment/lab.tex}
\input{statalign/lab.tex}
\input{rates/lab.tex}
\input{rnasingle/lab.tex}
\input{rnapair/lab.tex}
\input{rnatree/lab.tex}
\input{lineage/lab.tex}
\input{recombination/lab.tex}
\input{logic/lab.tex}
\input{space/lab.tex}
\input{diffusion/lab.tex}

% Bibliography
\bibliography{../../latex-inputs/alignment,../../latex-inputs/ncrna,../../latex-inputs/genomics,../../latex-inputs/reconstruction,../../latex-inputs/transposon,../../latex-inputs/duplication}

\end{document}
